### PR TITLE
[SCB-1096]change the method calculate process cpu rate to same with top

### PR DESCRIPTION
**Pull request by heyile**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean install` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

File: metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/meter/os/cpu/AbstractCpuUsage.java [hunk 1]
```diff
@@ -68,36 +62,21 @@ public void setUsage(double usage) {
...
-
-  protected String[] readAndSplitStat() throws IOException {
-    File file = new File(filePath);
-    String stat = FileUtils.readLines(file, StandardCharsets.UTF_8).get(0);
-    return stat.trim().split("\\s+");
-  }
-
-  public void update() {
-    String[] stats;
+  @SuppressWarnings("deprecation")
```
Comment (liubao68) [inline metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/meter/os/cpu/AbstractCpuUsage.java]: Do not use deprecated method, we can write a new one if no other available

Comment (heyile) [reply metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/meter/os/cpu/AbstractCpuUsage.java]: ok

File: metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/meter/os/cpu/OsCpuUsage.java [hunk 2]
```diff
@@ -19,48 +19,52 @@
...
-  private long readCurrentTotalTime(String[] stats) {
-    long total = 0L;
-    for (int i = 1; i <= 8; i++) {
-      total += Long.parseLong(stats[i]);
+  private void update(String[] stats) {
+
+    long currentIdle = Long.parseLong(stats[4]);
+    idle.update(currentIdle);
+
+    long totalCpu = 0L;
```
Comment (wujimin) [inline metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/meter/os/cpu/OsCpuUsage.java]: 1.delete useless empty lines.
2.why not just invoke CpuUtils.summary?

    String[] stats = CpuUtils.readAndSplitFirstLine(PROC_STAT);
    total.update(CpuUtils.summary(stats, 1, 8));
    idle.update(Double.parseDouble(stats[4]));
    super.updateUsage(total.period - idle.period, total.period, false);

File: metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/meter/os/cpu/OsCpuUsage.java [hunk 3]
```diff
@@ -19,48 +19,52 @@
...
   }
 
   @Override
-  protected long readCurrentBusyTime(String[] stats) {
-    return currentTotalTime - Long.parseLong(stats[4]);
-  }
-
-  public long getLastTotalTime() {
-    return lastTotalTime;
+  protected void updateUsage(double periodBusy, double periodTotal) {
```
Comment (wujimin) [inline metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/meter/os/cpu/OsCpuUsage.java]: move to parent class
```
  protected void updateUsage(double periodBusy, double periodTotal, boolean irixMode) {
    usage = periodTotal == 0 ? 0 : periodBusy / periodTotal;
    if (usage > 1) {
      usage = 1;
    }
    if (irixMode) {
      usage *= cpuCount;
    }
  }
```

File: metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/meter/os/cpu/ProcessCpuUsage.java [hunk 4]
```diff
@@ -16,48 +16,51 @@
...
-    throw new IllegalStateException("Failed to get current process Id");
+    updateUsage(busy.period, total.period);
   }
 
   @Override
-  protected long readCurrentBusyTime(String[] stats) {
-    long busy = 0L;
-    for (int i = 13; i <= 16; i++) {
-      busy += Long.parseLong(stats[i]);
+  protected void updateUsage(double periodBusy, double periodTotal) {
```
Comment (wujimin) [inline metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/meter/os/cpu/ProcessCpuUsage.java]: move to parent class

File: metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/meter/os/cpu/ProcessCpuUsage.java [hunk 5]
```diff
@@ -16,48 +16,51 @@
...
   }
 
-  private static String getCurrentPid() {
-    String name = ManagementFactory.getRuntimeMXBean().getName();
-    int idx = name.indexOf('@');
-    if (idx > 0) {
-      return name.substring(0, idx);
+  public void update() {
+    String[] stats = CpuUtils.readAndSplitFirstLine(CpuUtils.SELF_PROCESS);
+    String[] uptime = CpuUtils.readAndSplitFirstLine(CpuUtils.UPTIME);
```
Comment (wujimin) [inline metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/meter/os/cpu/ProcessCpuUsage.java]: why not just invoke readUptimeTotal?

File: metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/publish/DefaultLogPublisher.java [hunk 6]
```diff
@@ -176,7 +175,8 @@ private void printCpuLog(StringBuilder sb, MeasurementNode osNode) {
     double processRate = processNode.summary();
 
     appendLine(sb, "  cpu:");
-    appendLine(sb, "    all: %.2f%%    process: %.2f%%", allRate * 100, processRate * 100);
+    appendLine(sb, "    all: %.2f%%    process: %.2f%%    idle: %.2f%%", allRate * 100, processRate * 100,
+        (1 - allRate) * 100);
```
Comment (wujimin) [inline metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/publish/DefaultLogPublisher.java]: appendLine(sb, "    all usage: %.2f%%    all idle: %.2%%    process: %.2f%%", allRate * 100, (1 - allRate) * 100, processRate * 100);

**Pull request merged**